src/mdtest.c

```diff
@@ -2364,26 +2057,31 @@ mdtest_results_t * mdtest_run(int argc, char **argv, MPI_Comm world_com, FILE *
     /*   if directory does not exist, create it */
     if ((rank < path_count) && backend->access(testdirpath, F_OK, &param) != 0) {
         if (backend->mkdir(testdirpath, DIRMODE, &param) != 0) {
-            FAIL("Unable to create test directory path");
+            FAIL("Unable to create test directory path %s", testdirpath);
         }
     }
 
     /* display disk usage */
-    if (verbose >= 3 && rank == 0) {
-        fprintf(out_logfile,  "V-3: main (before display_freespace): testdirpath is \"%s\"\n", testdirpath );
-        fflush( out_logfile );
-    }
+    VERBOSE(3,-1,"main (before display_freespace): testdirpath is '%s'", testdirpath );
 
     if (rank == 0) display_freespace(testdirpath);
+    int tasksBlockMapping = QueryNodeMapping(testComm);
 
-    if (verbose >= 3 && rank == 0) {
-        fprintf(out_logfile,  "V-3: main (after display_freespace): testdirpath is \"%s\"\n", testdirpath );
-        fflush( out_logfile );
+    /* set the shift to mimic IOR and shift by procs per node */
+    if (nstride > 0) {
```

Comment:
Ideally the semantics surrounding the mdtest `-N` argument and the IOR `-C` argument would be similar, but they currently are not.  It makes me a little uncomfortable to make the behavior of `-N 1` depend on the way the MPI wireup maps ranks to processes, but I don't know a reasonable way to fix this.

That said, the help line in line 1865 should be updated to clarify that `-N 1` is not always a 1-task shift, but a 1-node shift when the resource manager does block allocation.

Reply:
I think that the semantics are almost the same now.  -N 1 in mdtest is identical to -C in IOR.  I agree it would be better if they were identical.  It would be great if it was the same flag but -N is already used for something else in IOR and -C is already used for something else in mdtest.  We could remove the argument to -N in mdtest; that would actually make a lot of sense since I think the only reason it was there was to allow users to explicitly set it to PPN for block allocation.  Want me to do that?

Reply:
Changed the help line.  Thanks!